Add tutorial for "Setting up a Java project" in VS Code

Johnwz123 opened the pull request.

damithc commented on index.md:
```diff
@@ -36,6 +36,9 @@ Tutorials, how-to guides, getting-started guides, ...
   * [Using the debugger](tutorials/intellijDebugger.html)
   * [Useful settings](tutorials/intellijUsefulSettings.html)
 * [**Visual Studio Code**](tutorials/vscode.html)
+  * [Setting up a Java project](tutorials/vscodeJavaSetup.html)
```
It's more natural to start with `Preparing VS Code for Java` section, followed by `Creating a new Java project`, and `Importing an existing project`?

Johnwz123 replied:
I agree. I have updated to split the page into separate pages for each section.

damithc commented on tutorials/vscodeJavaSetup.md:
```diff
@@ -0,0 +1,182 @@
+
+This tutorial is adapted from the [official VS Code Java Tutorial](https://code.visualstudio.com/docs/java/java-tutorial).
+</box>
+
```
Start by mentioning the installation of these first? Best not to assume they are installed already. 
1. JDK (point to our own installation instructions)
2. VS Code

damithc commented on tutorials/vscodeJavaSetup.md:
```diff
@@ -0,0 +1,182 @@
+
+## Install the Extension Pack for Java
+
+To develop Java applications in VS Code, you need to install the appropriate extensions that provide language support, debugging, and other essential features. The easiest way to get started with Java development in VS Code is to install the [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack), which includes all the essential extensions you need ...
```
Mention how to deal with this screen (given we recommend students to install the JDK first, they should choose `Mark Done`):
![image](https://github.com/user-attachments/assets/4e5ea827-39a7-47c7-866e-bd1bf144b138)

damithc commented on tutorials/vscodeJavaSetup.md:
```diff
@@ -0,0 +1,182 @@
+To develop Java applications in VS Code, you need to install the appropriate extensions that provide language support, debugging, and other essential features. The easiest way to get started with Java development in VS Code is to install the [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack), which includes all the essential extensions you need ...
+
+### Installation steps
+
```
Start with a brief explanation of what is an 'extension pack', using the Java extension pack as an example? i.e., explain the WHAT aspect before the HOW aspect.

Johnwz123 replied:
I've updated the writeup for the section.

damithc commented on tutorials/vscodeJavaSetup.md:
```diff
@@ -0,0 +1,182 @@
+### What's Included
+
+The [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack) includes these essential tools for Java development:
+* **Language Support for Java** - Syntax highlighting, IntelliSense, and error detection
+* **Debugger for Java** - Set breakpoints and step through your code
+* **Test Runner for Java** - Run and debug JUnit tests
+* **Maven for Java** - Support for Maven projects
+* **Gradle for Java** - Support for Gradle projects
+* **Project Manager for Java** - Manage Java projects and dependencies
+* **IntelliCode** - AI-assisted code completion
```
Probably no need to list all these? The extension page also has the same info.

damithc commented on tutorials/vscodeJavaSetup.md:
```diff
@@ -0,0 +1,182 @@
+
+<box type="tip" seamless>
+
+**Quick tip:** After installation, open the Command Palette: {{ icon_windows }}/{{ icon_linux}} `Ctrl+Shift+P` | {{ icon_apple }} `Cmd+Shift+P`, type `Java: Tips for Beginners` and open it for helpful information to get started.
```
Give screenshot? Apparently, it expects a `>` at the start.
![image](https://github.com/user-attachments/assets/cf5fc7bb-1940-4954-b813-a6a859d71830)

damithc commented on tutorials/vscodeJavaSetup.md:
```diff
@@ -0,0 +1,182 @@
+
+After installing the extensions, you should see:
+
+1. **Java language support** - `.java` files will have syntax highlighting
```
How to do this? At this point there is no .java file open in the IDE

damithc commented on tutorials/vscodeJavaSetup.md:
```diff
@@ -0,0 +1,182 @@
+After installing the extensions, you should see:
+
+1. **Java language support** - `.java` files will have syntax highlighting
+2. **Java commands** in the Command Palette - Open the Command Palette: {{ icon_windows }}/{{ icon_linux}} `Ctrl+Shift+P` | {{ icon_apple }} `Cmd+Shift+P` and type `Java:` to see available commands
```
Give screenshot?
![image](https://github.com/user-attachments/assets/8f947dfb-8c4a-4065-aa94-7f405191d411)

Johnwz123 replied:
Have added a screenshot.

damithc commented on tutorials/vscodeJavaSetup.md:
```diff
@@ -0,0 +1,182 @@
+
+1. **Java language support** - `.java` files will have syntax highlighting
+2. **Java commands** in the Command Palette - Open the Command Palette: {{ icon_windows }}/{{ icon_linux}} `Ctrl+Shift+P` | {{ icon_apple }} `Cmd+Shift+P` and type `Java:` to see available commands
+3. **Status bar indicators** - Java information in the bottom status bar
```
This wont work unless there is a java file open.

damithc commented on tutorials/vscodeJavaSetup.md:
```diff
@@ -0,0 +1,182 @@
+
+Before you can develop Java applications in VS Code, you need to have a Java Development Kit (JDK) installed and properly configured.
+
+### Install JDK
+
+VS Code can help you install and configure a JDK automatically through its integrated tools.
+
+1. **Open the Command Palette**: {{ icon_windows }}/{{ icon_linux}} `Ctrl+Shift+P` | {{ icon_apple }} `Cmd+Shift+P`
+2. **Type** `Java: Install New JDK` and select it
+3. **Choose a JDK version** (Java 17 or later recommended)
+4. **Follow the installation prompts**
+5. **Wait for the download and installation** to complete
```
I'd rather they install the JDK outside of VSC, to learn JDK installation independently from the IDE, and to ensure the precise version we want is installed.

damithc commented on tutorials/vscodeJavaSetup.md:
```diff
@@ -0,0 +1,182 @@
+
+1. **Open the Command Palette**: {{ icon_windows }}/{{ icon_linux}} `Ctrl+Shift+P` | {{ icon_apple }} `Cmd+Shift+P`
+2. **Type** `Java: Configure Java Runtime` and select it
+3. **Verify your JDK** appears in the dropdown list and is selected
```
The following error happens if there is no Java project open at this point:
![image](https://github.com/user-attachments/assets/1e77c7e2-8bdb-44b8-bcd1-5817ff6b2a2b)

Johnwz123 replied:
I've shifted the content in this section to the project setup pages.

damithc reviewed (changes requested):
Good start, @Johnwz123 
I reviewed only the first part. You can update that part first. Once we are happy with that part, we can revise the rest.

damithc commented:
@Johnwz123 Let me know when this is ready for review again.

Johnwz123 commented:
@damithc Have updated the PR accordingly based on your comments, mainly:

* Separated the sections into 3 separate pages for "Preparing VS Code for Java", "Creating a new Java project" and "Importing an existing project"
* Updated the contents accordingly
* Added screenshots for greater clarity

Additionally, as mentioned in the reply to your comment above. Would require your advice on the following:

Currently, for the "Install JDK" section, I've added instructions directing them to the Java Installation guide to install the JDK at the start of the section and shifted the steps for installing through VS Code as an alternative to the bottom of the section so that readers will be more likely to go to the Java installation guides first.

Would need your opinion on whether we should:

1. Include the alternative installation steps through VS Code - So that the guide would be more comprehensive
2. Exclude the alternative installation steps - To ensure students install the JDK outside of VS Code. In this case, the "Install JDK" section can also be shifted to come first at the top of the page, before the "Install VS Code" section so that students will install the JDK first before setting up VS Code if they follow the page flow.

![image](https://github.com/user-attachments/assets/9dd684d4-1bc0-4395-a314-ae78d28d4fff)

damithc commented:
@Johnwz123 something related: should we also include a mention of how to set configure VSC so that one can open a file in it by typing `code <file path>` into the terminal?

Johnwz123 commented:
> @Johnwz123 something related: should we also include a mention of how to set configure VSC so that one can open a file in it by typing `code <file path>` into the terminal?

Yes, I think that would be good. I'll add it as an update in my next PR.
